### PR TITLE
[FLINK-16538][python][docs] Restructure Python Table API documentation

### PR DESCRIPTION
**Pull request by dianfu**
## What is the purpose of the change

*This pull request restructures the Python Table API documentation.*

## Brief change log

  - *It add an item "Python Table API" under "Table API" and move the documentation about Python Table API under it.*

## Verifying this change

This change is a documentation work without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: docs/dev/table/python/index.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+title: "Python Table API"
+nav-id: python_tableapi
+nav-parent_id: tableapi
+nav-pos: 100
```
[the remark this answers is not in the document]
Comment (dianfu) [reply docs/dev/table/python/index.md]: I'm more tend to the current location for the doc of "Python Table API" for the following reasons:
- The documentation about "Table API" is about the operations in the Table class. However, the documentation about "Python Table API" is more general, it includes everything about the Python language support about the table module. So I think there is no direct relationship between "Table API" and "Python Table API". Maybe we should rename it to something else?
- The documentation about "Python Table API" is about the Python language support about the table module and each of the items at the beginning is about one specific functionality about table and so personally I think putting "Python Table API" at the end makes more sense.

**Comment by dianfu** [comment]
@hequn8128 Thanks a lot for the review. Have updated the PR.

**Comment by sunjincheng121** [comment]
Hi @dianfu @hequn8128 I think would be great if we CP this docs refactor changes to release-1.10 branch, as many people read mostly the "stable" docs, not the "latest master" docs. :) 
What do you think?

**Comment by dianfu** [comment]
@sunjincheng121 Thanks for suggestion. Good advice. I will cherry-pick it to release-1.10.

**Pull request merged**